fictadvisor-api/src/v2/api/controllers/DateController.ts

```diff
@@ -15,28 +16,20 @@ export class DateController {
   ) {}
 
   @Get('/semesters')
-  @ApiOkResponse({
-    type: SemestersResponse,
-  })
-  @ApiQuery({
-    name: 'isFinished',
-    type: Boolean,
-    required: false,
+  @ApiEndpoint({
+    summary: 'Gets recent semesters',
+    documentation: DateDocumentation.SEMESTERS,
   })
   async getPreviousSemesters (
-    @Query('isFinished', ConvertToBooleanPipe) isFinished?
+    @Query('isFinished', ConvertToBooleanPipe) isFinished?,
   ) {
     return { semesters: await this.dateService.getPreviousSemesters(isFinished as boolean) };
   }
```

Comment:
- this line must look like this:
```suggestion
    return this.dateService.getPreviousSemesters(isFinished);
  }
```
- `dateService.getPreviousSemesters` must return such object: `{ semesters }`

Reply:
`dateService.getPreviousSemesters` can't return { semesters }, because it's cause such error: 

> src/v2/api/services/PollService.ts:264:21 - error TS2488: Type '{ semesters: (GetResult<{ year: number; semester: number; startDate: Date; endDate: Date; createdAt: Date; updatedAt: Date; }, unknown> & {})[]; }' must have a '[Symbol.iterator]()' method that returns an iterator.

Should I refactor PollService to fix this?

Reply:
yes, you can destructure property `semesters` of returned object in poll service